publish-module.psm1

```diff
@@ -435,21 +438,204 @@ function Publish-AspNetFileSystem{
         $publishArgs += $sharedArgs.ExtraArgs
 
         $command = '"{0}" {1}' -f (Get-MSDeploy),($publishArgs -join ' ')
-        Print-CommandString -msdeployPath (Get-MSDeploy) -msdeployParameters ($publishArgs -join ' ')
-        Execute-CommandString -command $command
+        $command | Print-CommandString
+        $command | Execute-CommandString
+    }
+}
+
+function Publish-AspNetDocker{
+    [cmdletbinding(SupportsShouldProcess=$true)]
+    param(
+        [Parameter(Mandatory = $true,Position=0,ValueFromPipeline=$true,ValueFromPipelineByPropertyName=$true)]
+        $publishProperties,
+        [Parameter(Mandatory = $true,Position=1,ValueFromPipelineByPropertyName=$true)]
+        $packOutput
+    )
+    process{
+        if($publishProperties){
+            
+            # find out the right Dockerfile
+            $projectFolder = $(get-childitem (Join-Path $packOutput approot\src))[0];
+            $dockerfileRoot = Join-Path $projectFolder.FullName Properties\PublishProfiles
+
+            $dockerfileRelPath = $publishProperties['DockerfileRelativePath']
+            if($dockerfileRelPath){
+                $dockerfilePath = Resolve-Path (Join-Path $dockerfileRoot $dockerfileRelPath)
```

Comment:
What is the path relative to? From what I can see here it will be relative to the working directory which may not be correct. If it's relative to some known folder you should follow the pattern below.

```
Push-Location
Set-Location <path>
Pop-Location
```